examples/demo13.py

```diff
@@ -0,0 +1,378 @@
+#    and/or other materials provided with the distribution.
+#
+"""
+Demo #13
+
+The thirteenth script in our tutorial about using Galsim in python scripts: examples/demo*.py.
+(This file is designed to be viewed in a window 100 characters wide.)
+
+This script currently doesn't have an equivalent demo*.yaml or demo*.json file.
+
+This script introduces non-idealities arising from NIR detectors, in particular those that will be
+observed and accounted for in the WFIRST surveys. Three such non-ideal effects are demonstrated, in
+the order in which they are introduced in the detectors:
+
+1) Reciprocity failure: Flux-dependent sensitivity of the detector.
+2) Non-linearity: Charge-dependent gain in converting from units of electrons to ADU.  Non-linearity
+   in some form is also relevant for CCDs in addition to NIR detectors.
+3) Interpixel capacitance: Influence of charge in a pixel on the voltage reading of neighboring
+   ones.
+
+The purpose of the demo is two-fold: (1) to show the effects of detector non-idealities on images
+from NIR detectors, and (2) to illustrate the full image generation process, including all sources
+of noise at appropriate stages.
+
+New features introduced in this demo:
+- galsim.makeCOSMOSCatalog(...)
+- galsim.makeCOSMOSObj(...)
+- Adding sky level and dark current.
+- image.addReciprocityFailure(exp_time, alpha, base_flux)
+- image.quantize()
+- image.applyNonlinearity(NLfunc, *args)
+- image.applyIPC(IPC_kernel, edge_treatment, fill_value, kernel_nonnegativity,
+                 kernel_normalization)
+- readnoise.setReadNoise(readnoise_level)
+- All functionality in the galsim.wfirst module.
+"""
+
+import numpy
+import sys, os
+import math
+import logging
+import time
+import galsim as galsim
+import galsim.wfirst as wfirst
+
+def main(argv):
+    # Where to find and output data.
+    path, filename = os.path.split(__file__)
+    datapath = os.path.abspath(os.path.join(path, "data/"))
+    outpath = os.path.abspath(os.path.join(path, "output/"))
+
+    # In non-script code, use getLogger(__name__) at module scope instead.
+    logging.basicConfig(format="%(message)s", level=logging.INFO, stream=sys.stdout)
+    logger = logging.getLogger("demo13")
+
+    # Initialize (pseudo-)random number generator.
+    random_seed = 1234567
+    rng = galsim.BaseDeviate(random_seed)
+
+    # Generate a Poisson noise model.
+    poisson_noise = galsim.PoissonNoise(rng) 
+    logger.info('Poisson noise model created.')
+
+    # Read in the WFIRST filters, setting an AB zeropoint appropriate for this telescope given its
+    # diameter and (since we didn't use any keyword arguments to modify this) using the typical
+    # exposure time for WFIRST images.
+    filters = wfirst.getBandpasses(AB_zeropoint=True)
+    # We care about the NIR imaging, not the prism and grism, so let's remove them from the dict
+    # containing the filters:
+    del filters['SNPrism']
+    del filters['BAO-Grism']
```

Comment:
Which raises the question: Should `wfirst.getBandpasses` even include these two?  I think probably not.